web/src/components/upload-indicator/upload-indicator.context.tsx

```diff
@@ -0,0 +1,45 @@
+import { FC, createContext, useMemo, useState } from 'react';
+
+export type OnProgressCallback = (progress: number) => void;
```

Comment:
Let's place this where it's only used, I don't see where we import it.

Reply:
addressed

Reply:
I meant that we can move this definition directly to the place where we use it like we did for `resetUploadProgressState` function.

```js
export interface UploadProgressTracker {
    progress: number;
    setProgress: (progress: number) => void;
    isUploaded: boolean;
    resetUploadProgressState: () => void;
}
```

Reply:
Checked